src/FacadeAccountCreation/FacadeAccountCreation.Core/Services/RoleManagement/RoleManagementService.cs

```diff
@@ -141,8 +141,12 @@ public async Task<DelegatedPersonNominatorModel> GetDelegatedPersonNominator(Gui
         _httpClient.DefaultRequestHeaders.Clear();
         _httpClient.DefaultRequestHeaders.Add(XEprUserHeader, userId.ToString());
         _httpClient.DefaultRequestHeaders.Add(XEprOrganisationHeader, organisationId.ToString());
+        var uriBuilder = new UriBuilder(_httpClient.BaseAddress)
```

Comment:
What's the advantage of using UriBuilder when the previous code is also correct & simpler?

Reply:
It's throwing "Change this code to not construct the URL from user-controlled data" issue from SonarQube10.4

Previous code leads to security vulnerabilities such as injection attacks to prevent this we should use safer methods to build URL.

Please have a look this documentation
https://vss-sonarqube-upgrade.azure.defra.cloud/project/issues?issueStatuses=OPEN%2CCONFIRMED&open=3b8efc25-d623-49e6-9c3e-a3a907fed910&id=epr-pom-web-api-gateway

@davidtonthat